### PR TITLE
KAFKA-3725: Update documentation with regards to XFS

### PR DESCRIPTION
**Pull request by toddpalino**
I've updated the ops documentation with information on using the XFS filesystem, based on LinkedIn's testing (and subsequent switch from EXT4).

I've also added some information to clarify the potential risk to the suggested EXT4 options (again, based on my experience with a multiple broker failure situation).

### CODE REVIEW

**Comment by harshach** [comment]
+1

File: docs/ops.html [hunk 1]
```diff
@@ -516,10 +516,22 @@
   <li>It automatically uses all the free memory on the machine
 </ul>
 
-<h4><a id="ext4" href="#ext4">Ext4 Notes</a></h4>
-Ext4 may or may not be the best filesystem for Kafka. Filesystems like XFS supposedly handle locking during fsync better. We have only tried Ext4, though.
-<p>
-It is not necessary to tune these settings, however those wanting to optimize performance have a few knobs that will help:
+<h4><a id="filesystems" href="#filesystems">Filesystem Selection</a></h4>
+<p>Kafka uses regular files on disk, and as such it has no hard dependency on a specific filesystem. The two filesystems which have the most usage, however, are EXT4 and XFS. Historically, EXT4 has had more usage, but recent improvements to the XFS filesystem have show it to have better performance characterisitics for Kafka's workload with no compromise in stability.</p>
```
Comment (dpkp) [inline docs/ops.html]: sp: 'have shown'

File: docs/ops.html [hunk 2]
```diff
@@ -516,10 +516,22 @@
...
+  <li>nobarrier: For underlying devices that have battery-backed cache, this option can provide a little more performance by disabling periodic write flushes. However, if the underlying device is well-behaved, it will report to the filesystem that it does not require flushes, and this option will have no effect.</li>
+</ul>
+<h5><a id="ext4" href="#ext4">EXT4 Notes</a></h5>
+EXT4 is a servicable choice of filesystem for the Kafka data directories, however getting the most performance out of it will require adjusting several mount options. In addition, these options are generally unsafe in a failure scenario, and will result in much more data loss and corruption. For a single broker failure, this is not much of a concern as the disk can be wiped and the replicas rebuilt from the cluster. In a multiple-failure scenario, such as a power outage, this can mean underlying filesystem (and therefore data) corruption that is not easily recoverable. The following options can be adjusted:
```
Comment (dpkp) [inline docs/ops.html]: sp: 'serviceable'

**Comment by dpkp** [comment]
nice writeup!

**Comment by ijuma** [comment]
Thanks for the PR, this is a nice improvement. LGTM.

I fixed a typo and a missing `/` in a couple of closing h5 tags and then merged to trunk and 0.10.0 branches.

**Pull request closed without merging**